python/cudf/cudf/core/column/column.py

```diff
@@ -1037,10 +1038,31 @@ def unique(self) -> ColumnBase:
         return drop_duplicates([self], keep="first")[0]
 
     def serialize(self) -> Tuple[dict, list]:
+        # data model:
+
+        # Serialization produces a nested metadata "header" and a flattened
+        # list of memoryviews/buffers that reference data (frames).  Each
+        # header advertises a frame_count slot (FIXME: dtype serialization
+        # doesn't do this right now) which indicates how many frames
+        # deserialization will consume. The class used to construct an
+        # object is named under the key "type-serialized" to match with
+        # Dask's serialization protocol (see
+        # distributed.protocol.serialize). Since column dtypes may either be
+        # cudf native or foreign some special-casing is required here for
+        # serialization.
+
         header: Dict[Any, Any] = {}
         frames = []
         header["type-serialized"] = pickle.dumps(type(self))
-        header["dtype"] = self.dtype.str
+        try:
+            dtype, dtype_frames = self.dtype.serialize()
+            dtype["frame_count"] = len(dtype_frames)
+            header["dtype"] = dtype
+            frames.extend(dtype_frames)
+            header["dtype-is-cudf-serialized"] = True
```

Comment:
Naming is hard: `dtype-is-cudf-native` ?

Reply:
`dtype-is-cudf-serialized` is great!